datameta/security/__init__.py

```diff
@@ -103,16 +103,43 @@ def check_password_by_hash(pw, hashed_pw):
 
 def hash_token(token):
     """Hash a token and return the unsalted hash"""
-    hashed_token =  hashlib.sha256(token.encode('utf-8')).hexdigest()
+    hashed_token = hashlib.sha256(token.encode('utf-8')).hexdigest()
     return hashed_token
 
 
+def register_failed_login_attempt(db, user):
+    """ Registers a failed login attempt and disables user if this has happened too often in the last hour."""
+    BLOCK_AFTER_N_FAILED_LOGINS = 5
+
+    now = datetime.now()
+    last_hour = now - timedelta(hours=1)
+    db.add(LoginAttempt(user_id=user.id, timestamp=now))
+
+    failed_logins = sum(
+        1
+        for attempt in db.query(LoginAttempt).filter(LoginAttempt.user_id == user.id).all()
+        if attempt.timestamp - last_hour <= timedelta(hours=1)
```

Comment:
I'm not sure what you need the `last_hour` variable here for 🤔 Can't this just be
```python
now - timestamp <= timedelta(hours=1)
```
?